StockTrader/src/main/java/com/restResource/StockTrader/controller/BuyTriggerController.java

```diff
@@ -0,0 +1,169 @@
+package com.restResource.StockTrader.controller;
+
+import com.restResource.StockTrader.entity.BuyTrigger;
+import com.restResource.StockTrader.entity.CommandType;
+import com.restResource.StockTrader.entity.logging.ErrorEventLog;
+import com.restResource.StockTrader.entity.logging.UserCommandLog;
+import com.restResource.StockTrader.repository.AccountRepository;
+import com.restResource.StockTrader.repository.BuyTriggerRepository;
+import com.restResource.StockTrader.service.BuyTriggerService;
+import com.restResource.StockTrader.service.LoggingService;
+import com.restResource.StockTrader.service.QuoteService;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+
+@RestController
+@RequestMapping(value = "/buyTrigger")
+public class BuyTriggerController {
+
+    private AccountRepository accountRepository;
+
+    private LoggingService loggingService;
+
+    private BuyTriggerRepository buyTriggerRepository;
+
+    private BuyTriggerService buyTriggerService;
+
+    public BuyTriggerController(
+            BuyTriggerRepository buyTriggerRepository,
+            LoggingService loggingService,
+            BuyTriggerService buyTriggerService,
+            AccountRepository accountRepository) {
+
+        this.buyTriggerRepository = buyTriggerRepository;
+        this.loggingService = loggingService;
+        this.buyTriggerService = buyTriggerService;
+        this.accountRepository = accountRepository;
+    }
+
+    @PostMapping(path = "/amount")
+    public @ResponseBody
+    HttpStatus createTriggerAmount(
+            @RequestParam String userId,
+            @RequestParam String stockSymbol,
+            @RequestParam int stockAmount,
```

Comment:
I believe this is the dollar value to buy, not the number of stocks

Reply:
Just confirmed by looking at workload files that this should be the dollar amount

Reply:
I think the flow of this function needs to be explained. Ill go over it after class.

Reply:
I understand what you were trying to do, but you've misunderstood what the amount parameter is for the BUY_AMOUNT command. amount is the dollar amount that the user wants to buy, not the number of stocks.

Reply:
Ah your right, my bad, ill fix those up